superset/views/datasource.py

```diff
@@ -47,13 +50,24 @@ def save(self):
                 .filter(orm_datasource.owner_class.id.in_(datasource["owners"]))
                 .all()
             )
+
+        column_names = [o["column_name"] for o in datasource["columns"]]
```

Comment:
I think this could be simplified to (which is more efficient for counting), 

```
from collections import Counter

duplicates = [
    name
    for name, count in Counter(
        [col["column_name"] for col in datasource["columns"]]
    ).items()
    if count > 1
]

if duplicates:
    ...
```